src/dev/i18n/tasks/extract_untracked_translations.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  ErrorReporter,
+  I18nConfig,
+  matchEntriesWithExctractors,
+  normalizePath,
+  readFileAsync,
+} from '..';
+
+function filterEntries(entries: string[], exclude: string[]) {
+  return entries.filter((entry: string) =>
+    exclude.every((excludedPath: string) => !normalizePath(entry).startsWith(excludedPath))
+  );
+}
+
+export async function extractUntrackedMessages({
+  path,
+  config,
+}: {
+  path?: string | string[];
+  config: I18nConfig;
+}) {
+  const inputPaths = Array.isArray(path) ? path : [path || './'];
+  const availablePaths = Object.values(config.paths);
+  const ignore = availablePaths.concat([
+    '**/build/**',
+    '**/webpackShims/**',
+    '**/__fixtures__/**',
+    'utilities/**',
+    'built_assets/**',
+    'docs/**',
+    'optimize/**',
+    'data/**',
+    '**/target/**',
+    'tasks/**',
+    '**/test/**',
+    '**/scripts/**',
+    'src/dev/**',
+    '**/target/**',
+    'plugins/canvas/**',
```

Comment:
I'm excluding canvas for now because it takes 10 minutes to check the files there. I'm not fully familiar with the canvas plugin so I left it at this for now.

Reply:
I guess we'd just need to ignore `x-pack/plugins/canvas/canvas_plugin` that is basically a Canvas `target`?